File: configlibrary/src/main/java/com/shmedo/core/enums/BreakAlarmStatus.java
package com.shmedo.core.enums;

/**
 * 断线报警器状态
 */
public enum  BreakAlarmStatus {

    QUERY(0),OPEN(1), CLOSE(2);

    private int status;
    BreakAlarmStatus(int status) {
        this.status = status;
    }
    public int toInt() {
        return status;
    }
    public static BreakAlarmStatus valueOf(int status) {
        switch (status) {
            case 0: return QUERY;
            case 1: return OPEN;
            case 2: return CLOSE;
            default: return OPEN;
        }
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/md_cmd/enums/BreakAlarmStatus.kt
package com.shmedo.lib.device.base.md_cmd.enums

/**
 * 断线报警器状态
 */
enum class BreakAlarmStatus(private val status: Int) {
    QUERY(0),
    OPEN(1),
    CLOSE(2);

    fun toInt(): Int {
        return status
    }

    companion object {
        fun value(status: Int): BreakAlarmStatus {
            return when (status) {
                0 -> QUERY
                1 -> OPEN
                2 -> CLOSE
                else -> OPEN
            }
        }
    }
}
